q2_composition/_ancombc.py

```diff
@@ -120,6 +120,25 @@ def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
                                  ' \n\n'
                                  ' column::value pair with a value that was'
                                  ' not found: "%s"' % i)
+
+            # check that IDs associated with chosen reference level(s) are
+            # present within the input table
+            level_value_idx = meta.index[meta[column] == level_value].tolist()
+            table_idx_list = table.index.tolist()
+
+            for id in level_value_idx:
+                if id not in table_idx_list:
```

Comment:
This isn't quite correct, just because a given ID isn't in the table, doesn't mean that exists no ID of a given level which _would_ be in the table.

Reply:
A better approach may be to take the intersection of the `level_value_idx` and `table.index` and if that was empty, then raise an error about the missing IDs